src/analysis/functions/mod.rs

```diff
@@ -0,0 +1,159 @@
+use r2api::structs::LSectionInfo;
+use petgraph::graph::NodeIndex;
+
+use frontend::radeco_containers::{RadecoFunction, RadecoModule};
+use middle::ir::MOpcode;
+use middle::ssa::ssastorage::{SSAStorage};
+use middle::ssa::ssa_traits::SSA;
+use middle::ssa::ssa_traits::SSAWalk;
+use middle::ssa::ssa_traits::{NodeType, NodeData};
+use middle::regfile::SubRegisterFile;
+
+use std::collections::HashSet;
+
+
+pub struct FuncArgumentAnalyzer<'a> {
+    registers: &'a SubRegisterFile,
+    sections: &'a [LSectionInfo],
+    arguments: Vec<NodeIndex>
+}
+
+pub struct FuncInfo {
+    arguments: Vec<NodeIndex>
+}
+
+/// Argument filters define what is semantically important as an argument
+/// and what is not. For example, in x86, the "ret" opcode turns into the
+/// following SSA form:
+///
+/// $2 (rip) = Load(mem, rsp)
+/// $3 (rsp) = $1 (rsp) + 0x08
+///
+/// In this case note that ret pops the value of the return address into the
+/// rip register and then decreases the stack size by the size of the address.
+///
+/// While [rsp] is technically an argument in this case, it holds the value
+/// of the return address. Conventionally, this is an implicit argument 
+/// generated by the compiler. Every function invocation will have it and it
+/// is not necessarily important. These sort of arguments may vary on an
+/// architecture to architecture, and compiler to compiler basis, so this trait
+/// allows for generic filtering to remove un-important arguments.
+pub trait ArgumentFilter {
+    fn is_important(ssa: &SSAStorage, ni: NodeIndex) -> bool;
+}
+
+/// Let's every argument through, no filtering applies
+struct NoArgumentFilter;
+impl ArgumentFilter for NoArgumentFilter {
+    fn is_important(_ssa: &SSAStorage, _ni: NodeIndex) -> bool {
+        true
+    }
+}
+
+struct X86ArgumentFilter;
+impl ArgumentFilter for X86ArgumentFilter {
+    fn is_important(ssa: &SSAStorage, ni: NodeIndex) -> bool {
+        let nd = ssa.node_data(ni).unwrap();
+        let operands = ssa.operands_of(ni);
+
+        let dst_registers = ssa.registers(ni);
+
+        // rax is conventionally used as at the return value register
+        if dst_registers.len() == 1 && dst_registers[0] == "rax" {
```

Comment:
Don't hardcode these, use the alias_info from the register profile to be more generic. As an example, `rip` has "PC" as alias.